### PR TITLE
replace microcomponent with nanocomponent

### PR DESCRIPTION
**Pull request by aks-**
I refactored this, replacing microcomponent with nanocomponent, as in https://github.com/datproject/dat-desktop/issues/446, I will do the others in another PR

### CODE REVIEW

File: elements/header.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 'use strict'
```
Comment (juliangruber) [inline elements/header.js]: somehow this broke the hamburger menu, it's not opening up any more for me

**Comment by juliangruber** [comment]
I had to tweet about this: https://twitter.com/juliangruber/status/952194559328518149 (couldn't find your twitter handle)

File: elements/header.js (hunk 1 above)
Comment (aks-) [reply elements/header.js]: Hm, I will take a look when I have time

**Comment by aks-** [comment]
@juliangruber thanks for the fantastic praise, I have also forgotten twitter handle, haven't used it in long time. Thank you again :)

File: elements/header.js (hunk 1 above)
Comment (aks-) [reply elements/header.js]: fixed by @martinheidegger in this commit https://github.com/datproject/dat-desktop/pull/450/commits/04697c070fbcc93a836187807e6634d0e0e8cfd2 yay!

**Comment by aks-** [comment]
Still requires some fix, I will do it when I have time and let you folks know. Currently overloaded at work...

**Comment by aks-** [comment]
@juliangruber updated PR, can you please check it?

**Comment by juliangruber** [comment]
side note: please try in the future not to rebase such huge PRs, it can be hard to tell what you have reviewed already and what not

File: elements/header.js [hunk 2]
```diff
@@ -103,7 +103,7 @@ HeaderElement.prototype.createElement = function (props) {
   return html`
       <header class="${header}">
         <div class="fr relative">
-          ${importButton.render({
+          ${importButton.createElement({
```
Comment (juliangruber) [inline elements/header.js]: this should still use `importButton.render`, `createElement` is a private api. See https://github.com/choojs/nanocomponent#usage

File: elements/table-row.js [hunk 3]
```diff
@@ -118,52 +118,44 @@ module.exports = Row
 function Row () {
   if (!(this instanceof Row)) return new Row()
   Nanocomponent.call(this)
+
+  this.parts = {
```
Comment (juliangruber) [inline elements/table-row.js]: `parts` is a bit generic, can we call this `partials` maybe? That term is known from the Ruby on Rails world

File: lib/datInfo.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+'use strict'
+var bytes = require('prettier-bytes')
+
+function datPeers (dat) {
+  return dat.network ? dat.network.connected : 'N/A'
+}
+
+module.exports = {
+  datPeers,
```
Comment (juliangruber) [inline lib/datInfo.js]: why not inline `datPeers` here too?

Comment (martinheidegger) [reply lib/datInfo.js]: `datPeers` is used in `datState`. If `datState` would use `this.datPeers` to access it the method could not be able to simply use it like: `var {datState} = require('./lib/datInfo')`

**Comment by aks-** [comment]
@juliangruber can we remove appveyor tests for windows? or maybe try fixing them by updating deps for hypercore, hyperdive?

**Comment by juliangruber** [comment]
Afaik hypercore/hyperdrive isn't breaking the tests on windows, there were UI issues instead

File: lib/datInfo.js (hunk 4 above)
Comment (juliangruber) [reply lib/datInfo.js]: +1

**Comment by aks-** [comment]
@juliangruber hm I see... I haven't got windows machine otherwise I could have tried fixing it...

btw would you want anything else changed in this PR?

**Comment by juliangruber** [comment]
Since windows was failing before, I'm fine with merging this as long as all on the mac / travis side still works. I'll give this a last read through and then will merge!

**Comment by aks-** [comment]
@juliangruber 
screen shot 2018-01-22 at 6 21 58 pm

works, downloaded successfully.

**Comment by martinheidegger** [comment]
React branch used, closing this.

**Pull request closed without merging**